backend/src/main/java/capstone/storage/backend/ItemController.java

```diff
@@ -25,16 +25,23 @@ public List<Item> getAllItems() {
     @PostMapping("{eanToFind}")
     @ResponseStatus(HttpStatus.CREATED)
     public Item saveItem(@PathVariable String eanToFind, @RequestBody AddItemDto addItemDto) {
+        if (service.isNullOrEmpty(addItemDto)) {
+            throw new ResponseStatusException(HttpStatus.METHOD_NOT_ALLOWED);
+        }
+
         if (addItemDto.ean().equals(eanToFind)) {
+
+
             try {
                 return service.addItem(addItemDto, eanToFind);
             } catch (ItemAlreadyExistException e) {
-                throw new ResponseStatusException(HttpStatus.BAD_REQUEST, e.getMessage());
+                throw new ResponseStatusException(HttpStatus.BAD_REQUEST);
+            } catch (IllegalArgumentException e) {
+                throw new ResponseStatusException(HttpStatus.PRECONDITION_FAILED);
             }
         }
         throw new ResponseStatusException(HttpStatus.FORBIDDEN);
```

Comment:
FORBIDDEN is indicating that a user is not authorised for the particular action. If that is the case here, it's fine. If not, consider using a different status